tests/api/test_enrolments.py

```diff
@@ -0,0 +1,224 @@
+from unittest import mock
+
+from fastapi.testclient import TestClient
+
+from app.domain.entities.enrolment import Enrolment
+from app.main import app
+from app.responses import FailureType, ResponseFailure, ResponseSuccess, SuccessType
+from tests.test_data.callback_provider import CallbackDataProvider
+from tests.test_data.enrolment_data_provider import DataProvider
+
+test_data = CallbackDataProvider()
+test_data_2 = DataProvider()
+client = TestClient(app)
+
+
+@mock.patch("app.use_cases.create_new_enrolment.CreateNewEnrolment")
+def test_create_enrolment_success_created(use_case):
+    code = SuccessType.CREATED
+    message = "The enrolment has been created."
+    use_case().execute.return_value = ResponseSuccess(
+        value=test_data.sample_enrolment,
+        type=code,
+        message=message,
+    )
+
+    data = test_data.sample_enrolment_request
+    response = client.post("/enrolments", data=data.json())
+    json_result = response.json()
+    enrolment = Enrolment(**json_result.get("value"))
+
+    use_case().execute.assert_called_with(data)
+    assert response.status_code == SuccessType.CREATED.value
+    assert enrolment == test_data.sample_enrolment
+    assert json_result.get("message") == message
+
+
+@mock.patch("app.use_cases.create_new_enrolment.CreateNewEnrolment")
+def test_create_enrolment_success(use_case):
```

Comment:
There is no code in use_cases.create_new_enrolment.CreateNewEnrolment for fetching enrolment. there is only for creating a one.